packages/ui/src/utils/toTree.ts

```diff
@@ -0,0 +1,34 @@
+import { AppQueue } from '@bull-board/api/typings/app';
+
+export interface AppQueueTree {
+  [key: string]: {
+    children: AppQueueTree;
+    queue?: AppQueue;
+  };
+}
+
+export function toTree(array: AppQueue[]): AppQueueTree {
+  const tree: AppQueueTree = {};
```

Comment:
There is no any enefit to use here an Object (it even deoptimizes the JS engine).

I think that It should be an array of objects that has children

It will be much simpler to traverse it